pre_commit/git.py

```diff
@@ -197,6 +197,11 @@ def commit(repo: str = '.') -> None:
     cmd_output_b(*cmd, cwd=repo, env=env)
 
 
+def update_changes(repo: str = '.') -> None:
+    cmd = ('git', 'add', '-u')
```

Comment:
What does this do if there were changes to the working set when we commit? Does `pre-commit` fully stash those before running the hooks?

What does this do if a different hook that doesn't set `commit_changes: true` does make changes to files?

I want to make sure we're only adding things to the index that we're confident should actually be added, and I think that might be a bit subtly tricky here.

Reply:
Yeah, from reading the docs + testing a bit it looks like `pre-commit` stashes all unstaged changes.

RE other hooks -- that's a good question: see other comment.

Reply:
Random thought - is it possible for a hook to make and commit changes that would then cause unstashing conflicts? What does that look like for the user?